### PR TITLE
Use hir::Node helper methods instead of repeating the same impl multiple times

### PR DESCRIPTION
**Pull request by oli-obk**
I wanted to do something entirely different and stumbled upon a bunch of cleanups

### CODE REVIEW

File: compiler/rustc_codegen_ssa/src/mir/block.rs [hunk 1]
```diff
@@ -852,7 +852,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         }
 
         let instance = match intrinsic {
-            None | Some(ty::IntrinsicDef { name: sym::drop_in_place, .. }) => instance,
+            None => instance,
```
Comment (compiler-errors) [inline compiler/rustc_codegen_ssa/src/mir/block.rs:855]: please move this commit to another pr 😆

Comment (oli-obk) [reply compiler/rustc_codegen_ssa/src/mir/block.rs]: done

**Review by compiler-errors — APPROVED** [review]
Sorry for taking so long on this, r=me after rebase
